pkg/kv/kvserver/batcheval/cmd_gc.go

```diff
@@ -37,22 +37,26 @@ func declareKeysGC(
 	_ time.Duration,
 ) {
 	gcr := req.(*roachpb.GCRequest)
-	// When GC-ing MVCC range key tombstones or individual range keys, we need to
-	// serialize with all writes that overlap the MVCC range tombstone, as well as
-	// the immediate left/right neighboring keys. This is because a range key
-	// write has non-local effects, i.e. it can fragment or merge other range keys
-	// at other timestamps and at its boundaries, and this has a non-commutative
-	// effect on MVCC stats -- if someone writes a new range key while we're GCing
-	// one below, the stats would come out wrong.
-	// Note that we only need to serialize with writers (including other GC
-	// processes) and not with readers (that are guaranteed to be above the GC
-	// threshold). To achieve this, we declare read-write access at
-	// hlc.MaxTimestamp which will not block any readers.
-	for _, span := range mergeAdjacentSpans(makeLookupBoundariesForGCRanges(
-		rs.GetStartKey().AsRawKey(), nil, gcr.RangeKeys,
-	)) {
-		latchSpans.AddMVCC(spanset.SpanReadWrite, span, hlc.MaxTimestamp)
+	if gcr.RangeKeys != nil {
+		// When GC-ing MVCC range key tombstones or individual range keys, we need
+		// to serialize with all range writes(deletes) that overlap the MVCC range
```

Comment:
nit: range key writes -- we specifically don't want to serialize with all range writes. Also, this latch will serialize with all MVCC range key writes to the entire range, not just those we overlap.

Reply:
Comments are hard! 😓